website/content/en/highlights/2022-05-18-vrl-iteration-support.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "VRL iteration support has arrived!"
+short: "VRL iteration"
+description: "Remap all your collections using iteration functions."
+authors: ["JeanMertz"]
+date: "2022-05-18"
+pr_numbers: [12317]
+release: "0.22.0"
+hide_on_release_notes: false
+badges:
+  type: announcement
+  domains: ["vrl", "remap"]
+tags: ["vector remap language", "vrl", "dsl", "iteration", "enumeration"]
+---
+
+Since we launched **Vector Remap Language** (VRL) [more than a year ago][], the
+language has seen a massive growth in both usage, and feature richness. While we
+intentionally left the core language in a stable state since its launch, we
+_have_ been steadily expanding the [standard library][], increasing
+[performance][], adding [event enrichment features][], and [squashing many
+bugs][].
+
+One of the last major pieces missing from the language — and one requested by
+many of you — is support for mapping collections. We are happy to announce that
+as of Vector 0.22.0, this feature is now available.
+
+Let’s take a look at iteration in VRL, and how it works.
+
+## Usage
+
+Let’s take a simple example:
+
+```json
+{
+  "foo": true,
+  "bar": { "bar": "" },
+  "baz": "",
+  "qux": ["a", "b", "a", "c"]
+}
+```
+
+Let’s assume we want to achieve the following results:
+
+1. Upcase all keys in the object.
+2. Change all empty string values to `null`.
+3. Count the number of equal elements in the `qux` array.
+
+Here’s how we would solve these individual tasks:
+
+```coffee
+# 1. Upcase all keys in the object.
+. = map_keys(., recursive: true) -> |key| { upcase(key) }
+
+# 2. Change all empty string values to `null`.
+. = map_values(., recursive: true) -> |value| {
+  if value == ”” { null } else { value }
+}
+
+# 3. Count the number of equal elements in the `qux` array.
+.qux_tally = {}
+for_each(.qux) -> |_index, value| {
+  tally = int(get!(.qux_tally, [value])) ?? 0
+
+  .qux_tally = set!(.qux_tally, [value], tally + 1)
+}
+```
+
+Running this VRL program results in the following output:
+
+```json
+{
+  "FOO": true,
+  "BAR": { "BAR": null },
+  "BAZ": null,
+  "QUX": ["a", "b", "a", "c"],
+  "qux_tally": {
+    "a": 2,
+    "b": 1,
+    "c": 1
+}
```

Comment:
```suggestion
    "c": 1
    }
}
```

This should also make the file diff coloring below happier.

Reply:
Resolved in 619ecdf2c.